src/sql/src/plan/statement/show.rs

```diff
@@ -407,25 +409,51 @@ fn show_sinks<'a>(
     scx: &'a StatementContext<'a>,
     full: bool,
     from: Option<UnresolvedObjectName>,
+    in_cluster: Option<RawIdent>,
     filter: Option<ShowStatementFilter<Raw>>,
 ) -> Result<ShowSelect<'a>, anyhow::Error> {
-    let schema = if let Some(from) = from {
-        scx.resolve_schema(from)?
-    } else {
-        scx.resolve_active_schema()?
+    let mut query_filters = vec![];
+
+    if let Some(from) = from {
+        query_filters.push(format!("schema_id = {}", scx.resolve_schema(from)?.id()));
+    } else if in_cluster.is_none() {
+        query_filters.push(format!("schema_id = {}", scx.resolve_active_schema()?.id()));
     };
 
+    if let Some(cluster) = in_cluster {
+        scx.require_experimental_mode("SHOW SINKS...IN CLUSTER")?;
+        query_filters.push(format!(
+            "clusters.id = {}",
+            resolve_names_cluster(scx, cluster)?.0
+        ));
+    }
+
+    let query_filters = itertools::join(query_filters.iter(), " AND ");
```

Comment:
👍🏽 we really should use a proper query builder at some point. Thanks for not adding to the `if` chain below in the meantime though!